core/src/robot/Robot.cc

```diff
@@ -474,6 +486,51 @@ namespace jiminy
         return returnCode;
     }
 
+
+    hresult_t Robot::addConstraint(std::string const & constraintName,
+                                   std::shared_ptr<AbstractConstraint> constraint)
+    {
+        hresult_t returnCode = hresult_t::SUCCESS;
+
+        if (constraintsHolder_.count(constraintName) > 0)
+        {
+            std::cout << "Error - Robot::addConstraint - A constraint with name " << constraintName;
+            std::cout << " already exists." << std::endl;
+            returnCode = hresult_t::ERROR_BAD_INPUT;
+        }
+        else
+        {
+            returnCode = constraint->initialize(this);
+            if (returnCode != hresult_t::SUCCESS)
+            {
+                std::cout << "Error - Robot::addConstraint - Fail to initialize constraint. " << std::endl;
+                returnCode = hresult_t::ERROR_BAD_INPUT;
+            }
+            else
+            {
+                constraintsHolder_.insert(std::make_pair(constraintName, constraint));
+            }
+        }
+        return returnCode;
+    }
+
+    hresult_t Robot::removeConstraint(std::string const & constraintName)
+    {
+        hresult_t returnCode = hresult_t::SUCCESS;
```

Comment:
no need for a returnCode variable here

Reply:
.

Reply:
I'm doing this for symmetry with other methods like detachMotor or detachSensors which work with the same principle of having only one return at the end.